In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")

In [2]:
df=pd.read_csv('googleplaystore.csv')
df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
App               10841 non-null object
Category          10841 non-null object
Rating            9367 non-null float64
Reviews           10841 non-null object
Size              10841 non-null object
Installs          10841 non-null object
Type              10840 non-null object
Price             10841 non-null object
Content Rating    10840 non-null object
Genres            10841 non-null object
Last Updated      10841 non-null object
Current Ver       10833 non-null object
Android Ver       10838 non-null object
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [4]:
df.shape

(10841, 13)

# Observations:
    10841 rows present in df
    13 columns present in df

In [5]:
df.describe()
# rating column is flot64 thats why 1 column stats anlaysis shown

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

In [6]:
df[df.duplicated()]

App      Category  Rating  \
229                       Quick PDF Scanner + OCR FREE      BUSINESS     4.2   
236                                                Box      BUSINESS     4.2   
239                                 Google My Business      BUSINESS     4.4   
256                                ZOOM Cloud Meetings      BUSINESS     4.4   
261                          join.me - Simple Meetings      BUSINESS     4.0   
...                                                ...           ...     ...   
8643                    Wunderlist: To-Do List & Tasks  PRODUCTIVITY     4.6   
8654   TickTick: To Do List with Reminder, Day Planner  PRODUCTIVITY     4.6   
8658                           ColorNote Notepad Notes  PRODUCTIVITY     4.6   
10049        Airway Ex - Intubate. Anesthetize. Train.       MEDICAL     4.3   
10768                                             AAFP       MEDICAL     3.8   

       Reviews                Size      Installs  Type Price Content Rating  \
229      80805  Varies with device    5,000,000+  Free     0       Everyone   
236     159872  Varies with device   10,000,000+  Free     0       Everyone   
239      70991  Varies with device    5,000,000+  Free     0       Everyone   
256      31614                 37M   10,000,000+  Free     0       Everyone   
261       6989  Varies with device    1,000,000+  Free     0       Everyone   
...        ...                 ...           ...   ...   ...            ...   
8643    404610  Varies with device   10,000,000+  Free     0       Everyone   
8654     25370  Varies with device    1,000,000+  Free     0       Everyone   
8658   2401017  Varies with device  100,000,000+  Free     0       Everyone   
10049      123                 86M       10,000+  Free     0       Everyone   
10768       63                 24M       10,000+  Free     0       Everyone   

             Genres       Last Updated         Current Ver         Android Ver  
229        Business  February 26, 2018  Varies with device        4.0.3 and up  
236        Business      July 31, 2018  Varies with device  Varies with device  
239        Business      July 24, 2018    2.19.0.204537701          4.4 and up  
256        Business      July 20, 2018      4.1.28165.0716          4.0 and up  
261        Business      July 16, 2018           4.3.0.508          4.4 and up  
...             ...                ...                 ...                 ...  
8643   Productivity      April 6, 2018  Varies with device  Varies with device  
8654   Productivity     August 6, 2018  Varies with device  Varies with device  
8658   Productivity      June 27, 2018  Varies with device  Varies with device  
10049       Medical       June 1, 2018              0.6.88          5.0 and up  
10768       Medical      June 22, 2018               2.3.1          5.0 and up  

[483 rows x 13 columns]

In [7]:
# change data type
df.describe(include= 'all').T

count unique                 top   freq     mean       std  \
App             10841   9660              ROBLOX      9      NaN       NaN   
Category        10841     34              FAMILY   1972      NaN       NaN   
Rating           9367    NaN                 NaN    NaN  4.19334  0.537431   
Reviews         10841   6002                   0    596      NaN       NaN   
Size            10841    462  Varies with device   1695      NaN       NaN   
Installs        10841     22          1,000,000+   1579      NaN       NaN   
Type            10840      3                Free  10039      NaN       NaN   
Price           10841     93                   0  10040      NaN       NaN   
Content Rating  10840      6            Everyone   8714      NaN       NaN   
Genres          10841    120               Tools    842      NaN       NaN   
Last Updated    10841   1378      August 3, 2018    326      NaN       NaN   
Current Ver     10833   2832  Varies with device   1459      NaN       NaN   
Android Ver     10838     33          4.1 and up   2451      NaN       NaN   

                min  25%  50%  75%  max  
App             NaN  NaN  NaN  NaN  NaN  
Category        NaN  NaN  NaN  NaN  NaN  
Rating            1    4  4.3  4.5   19  
Reviews         NaN  NaN  NaN  NaN  NaN  
Size            NaN  NaN  NaN  NaN  NaN  
Installs        NaN  NaN  NaN  NaN  NaN  
Type            NaN  NaN  NaN  NaN  NaN  
Price           NaN  NaN  NaN  NaN  NaN  
Content Rating  NaN  NaN  NaN  NaN  NaN  
Genres          NaN  NaN  NaN  NaN  NaN  
Last Updated    NaN  NaN  NaN  NaN  NaN  
Current Ver     NaN  NaN  NaN  NaN  NaN  
Android Ver     NaN  NaN  NaN  NaN  NaN

In [8]:
df['Reviews'].head()

0       159
1       967
2     87510
3    215644
4       967
Name: Reviews, dtype: object

In [10]:
df['Reviews'].shape

(10841,)

In [11]:
df['Reviews'].dtype

dtype('O')

In [12]:
df.Reviews.str.isnumeric().sum()

10840

# observation:
    Reviews dtype is object
    Reviews 10840 rows contain numeric value and 1 row contain string value

In [9]:
df['Reviews'].head()

0       159
1       967
2     87510
3    215644
4       967
Name: Reviews, dtype: object

In [95]:
df[~df['Reviews'].str.isnumeric()]
# finding in Reviews which row contain string value

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

In [96]:
df_copy=df.copy()
# make copy of df it denoted as df_copy

In [97]:
df_copy=df_copy.drop(df_copy.index[10472])
df_copy.head()
# drop string value row

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [98]:
df_copy['Reviews'].dtype

dtype('O')

In [102]:
df_copy['Reviews']=df_copy['Reviews'].astype('int')
# convert Reviews datatype object to integer

In [104]:
df_copy['Size'].unique()
# find out unique value

array(['19M', '14M', '8.7M', '25M', '2.8M', '5.6M', '29M', '33M', '3.1M',
       '28M', '12M', '20M', '21M', '37M', '2.7M', '5.5M', '17M', '39M',
       '31M', '4.2M', '7.0M', '23M', '6.0M', '6.1M', '4.6M', '9.2M',
       '5.2M', '11M', '24M', 'Varies with device', '9.4M', '15M', '10M',
       '1.2M', '26M', '8.0M', '7.9M', '56M', '57M', '35M', '54M', '201k',
       '3.6M', '5.7M', '8.6M', '2.4M', '27M', '2.5M', '16M', '3.4M',
       '8.9M', '3.9M', '2.9M', '38M', '32M', '5.4M', '18M', '1.1M',
       '2.2M', '4.5M', '9.8M', '52M', '9.0M', '6.7M', '30M', '2.6M',
       '7.1M', '3.7M', '22M', '7.4M', '6.4M', '3.2M', '8.2M', '9.9M',
       '4.9M', '9.5M', '5.0M', '5.9M', '13M', '73M', '6.8M', '3.5M',
       '4.0M', '2.3M', '7.2M', '2.1M', '42M', '7.3M', '9.1M', '55M',
       '23k', '6.5M', '1.5M', '7.5M', '51M', '41M', '48M', '8.5M', '46M',
       '8.3M', '4.3M', '4.7M', '3.3M', '40M', '7.8M', '8.8M', '6.6M',
       '5.1M', '61M', '66M', '79k', '8.4M', '118k', '44M', '695k', '1.6M',
     

In [106]:
for i in df_copy['Size'].str:
    print(i)

0        1
1        1
2        8
3        2
4        2
        ..
10836    5
10837    3
10838    9
10839    V
10840    1
Name: Size, Length: 10840, dtype: object
0        9
1        4
2        .
3        5
4        .
        ..
10836    3
10837    .
10838    .
10839    a
10840    9
Name: Size, Length: 10840, dtype: object
0        M
1        M
2        7
3        M
4        8
        ..
10836    M
10837    6
10838    5
10839    r
10840    M
Name: Size, Length: 10840, dtype: object
0        NaN
1        NaN
2          M
3        NaN
4          M
        ... 
10836    NaN
10837      M
10838      M
10839      i
10840    NaN
Name: Size, Length: 10840, dtype: object
0        NaN
1        NaN
2        NaN
3        NaN
4        NaN
        ... 
10836    NaN
10837    NaN
10838    NaN
10839      e
10840    NaN
Name: Size, Length: 10840, dtype: object
0        NaN
1        NaN
2        NaN
3        NaN
4        NaN
        ... 
10836    NaN
10837    NaN
10838    NaN
10839      s
10840    NaN
Nam

In [111]:
df_copy['Size']=df_copy['Size'].str.replace('M','000')
# replace all M into '000'

In [112]:
df_copy['Size']

0         19000
1         14000
2        8.7000
3         25000
4        2.8000
          ...  
10836     53000
10837    3.6000
10838    9.5000
10839       nan
10840     19000
Name: Size, Length: 10840, dtype: object

In [113]:
df_copy['Size']=df_copy['Size'].str.replace('k','')
# replace all k to ''

In [114]:
df_copy['Size'].unique()
# removing M,k


array(['19000', '14000', '8.7000', '25000', '2.8000', '5.6000', '29000',
       '33000', '3.1000', '28000', '12000', '20000', '21000', '37000',
       '2.7000', '5.5000', '17000', '39000', '31000', '4.2000', '7.0000',
       '23000', '6.0000', '6.1000', '4.6000', '9.2000', '5.2000', '11000',
       '24000', 'nan', '9.4000', '15000', '10000', '1.2000', '26000',
       '8.0000', '7.9000', '56000', '57000', '35000', '54000', '201',
       '3.6000', '5.7000', '8.6000', '2.4000', '27000', '2.5000', '16000',
       '3.4000', '8.9000', '3.9000', '2.9000', '38000', '32000', '5.4000',
       '18000', '1.1000', '2.2000', '4.5000', '9.8000', '52000', '9.0000',
       '6.7000', '30000', '2.6000', '7.1000', '3.7000', '22000', '7.4000',
       '6.4000', '3.2000', '8.2000', '9.9000', '4.9000', '9.5000',
       '5.0000', '5.9000', '13000', '73000', '6.8000', '3.5000', '4.0000',
       '2.3000', '7.2000', '2.1000', '42000', '7.3000', '9.1000', '55000',
       '23', '6.5000', '1.5000', '7.5000', '51000'

In [115]:
# replace Varies with device by nan but here all numbers in string thats why here write str(np.nan)
df_copy['Size']=df_copy['Size'].str.replace('Varies with device',str(np.nan))
df_copy['Size']

0         19000
1         14000
2        8.7000
3         25000
4        2.8000
          ...  
10836     53000
10837    3.6000
10838    9.5000
10839       nan
10840     19000
Name: Size, Length: 10840, dtype: object

In [117]:
df_copy['Size'].unique()

array(['19000', '14000', '8.7000', '25000', '2.8000', '5.6000', '29000',
       '33000', '3.1000', '28000', '12000', '20000', '21000', '37000',
       '2.7000', '5.5000', '17000', '39000', '31000', '4.2000', '7.0000',
       '23000', '6.0000', '6.1000', '4.6000', '9.2000', '5.2000', '11000',
       '24000', 'nan', '9.4000', '15000', '10000', '1.2000', '26000',
       '8.0000', '7.9000', '56000', '57000', '35000', '54000', '201',
       '3.6000', '5.7000', '8.6000', '2.4000', '27000', '2.5000', '16000',
       '3.4000', '8.9000', '3.9000', '2.9000', '38000', '32000', '5.4000',
       '18000', '1.1000', '2.2000', '4.5000', '9.8000', '52000', '9.0000',
       '6.7000', '30000', '2.6000', '7.1000', '3.7000', '22000', '7.4000',
       '6.4000', '3.2000', '8.2000', '9.9000', '4.9000', '9.5000',
       '5.0000', '5.9000', '13000', '73000', '6.8000', '3.5000', '4.0000',
       '2.3000', '7.2000', '2.1000', '42000', '7.3000', '9.1000', '55000',
       '23', '6.5000', '1.5000', '7.5000', '51000'

In [53]:
df_copy['Size'].dtype

dtype('O')

In [118]:
df_copy["Size"]=df_copy["Size"].astype('float')
df_copy["Size"]
# converting object to float byusing astype

0        19000.0
1        14000.0
2            8.7
3        25000.0
4            2.8
          ...   
10836    53000.0
10837        3.6
10838        9.5
10839        NaN
10840    19000.0
Name: Size, Length: 10840, dtype: float64

In [119]:
df_copy['Size'].head()

0    19000.0
1    14000.0
2        8.7
3    25000.0
4        2.8
Name: Size, dtype: float64

In [120]:
df_copy['Size'][2]*1000

8700.0

In [57]:
for i in df_copy['Size']:
    if i <10:
        df_copy['Size']=df_copy['Size'].replace(i,i*1000)

In [121]:
df_copy['Size']=df_copy['Size']/1000
df_copy['Size']

0        19.0000
1        14.0000
2         0.0087
3        25.0000
4         0.0028
          ...   
10836    53.0000
10837     0.0036
10838     0.0095
10839        NaN
10840    19.0000
Name: Size, Length: 10840, dtype: float64

In [62]:
df_copy.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [63]:
df_copy.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

   Reviews    Size     Installs  Type Price Content Rating  \
0      159  0.0190      10,000+  Free     0       Everyone   
1      967  0.0140     500,000+  Free     0       Everyone   
2    87510  0.0087   5,000,000+  Free     0       Everyone   
3   215644  0.0250  50,000,000+  Free     0           Teen   
4      967  0.0028     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [65]:
df_copy['Installs']

0            10,000+
1           500,000+
2         5,000,000+
3        50,000,000+
4           100,000+
            ...     
10836         5,000+
10837           100+
10838         1,000+
10839         1,000+
10840    10,000,000+
Name: Installs, Length: 10840, dtype: object

In [66]:
df_copy['Installs'].dtype

dtype('O')

In [69]:
df_copy['Installs'].unique()

array(['10,000+', '500,000+', '5,000,000+', '50,000,000+', '100,000+',
       '50,000+', '1,000,000+', '10,000,000+', '5,000+', '100,000,000+',
       '1,000,000,000+', '1,000+', '500,000,000+', '50+', '100+', '500+',
       '10+', '1+', '5+', '0+', '0'], dtype=object)

In [68]:
df_copy['Price'].unique()

array(['0', '$4.99', '$3.99', '$6.99', '$1.49', '$2.99', '$7.99', '$5.99',
       '$3.49', '$1.99', '$9.99', '$7.49', '$0.99', '$9.00', '$5.49',
       '$10.00', '$24.99', '$11.99', '$79.99', '$16.99', '$14.99',
       '$1.00', '$29.99', '$12.99', '$2.49', '$10.99', '$1.50', '$19.99',
       '$15.99', '$33.99', '$74.99', '$39.99', '$3.95', '$4.49', '$1.70',
       '$8.99', '$2.00', '$3.88', '$25.99', '$399.99', '$17.99',
       '$400.00', '$3.02', '$1.76', '$4.84', '$4.77', '$1.61', '$2.50',
       '$1.59', '$6.49', '$1.29', '$5.00', '$13.99', '$299.99', '$379.99',
       '$37.99', '$18.99', '$389.99', '$19.90', '$8.49', '$1.75',
       '$14.00', '$4.85', '$46.99', '$109.99', '$154.99', '$3.08',
       '$2.59', '$4.80', '$1.96', '$19.40', '$3.90', '$4.59', '$15.46',
       '$3.04', '$4.29', '$2.60', '$3.28', '$4.60', '$28.99', '$2.95',
       '$2.90', '$1.97', '$200.00', '$89.99', '$2.56', '$30.99', '$3.61',
       '$394.99', '$1.26', '$1.20', '$1.04'], dtype=object)

In [122]:
chars_to_remove=['+',',','$']
cols_to_clean=['Installs','Price']
for item in chars_to_remove:
    for col in cols_to_clean:
        df_copy[col]=df_copy[col].str.replace(item,'')

In [123]:
df_copy['Price'].unique()

array(['0', '4.99', '3.99', '6.99', '1.49', '2.99', '7.99', '5.99',
       '3.49', '1.99', '9.99', '7.49', '0.99', '9.00', '5.49', '10.00',
       '24.99', '11.99', '79.99', '16.99', '14.99', '1.00', '29.99',
       '12.99', '2.49', '10.99', '1.50', '19.99', '15.99', '33.99',
       '74.99', '39.99', '3.95', '4.49', '1.70', '8.99', '2.00', '3.88',
       '25.99', '399.99', '17.99', '400.00', '3.02', '1.76', '4.84',
       '4.77', '1.61', '2.50', '1.59', '6.49', '1.29', '5.00', '13.99',
       '299.99', '379.99', '37.99', '18.99', '389.99', '19.90', '8.49',
       '1.75', '14.00', '4.85', '46.99', '109.99', '154.99', '3.08',
       '2.59', '4.80', '1.96', '19.40', '3.90', '4.59', '15.46', '3.04',
       '4.29', '2.60', '3.28', '4.60', '28.99', '2.95', '2.90', '1.97',
       '200.00', '89.99', '2.56', '30.99', '3.61', '394.99', '1.26',
       '1.20', '1.04'], dtype=object)

In [124]:
df_copy['Installs'].unique()

array(['10000', '500000', '5000000', '50000000', '100000', '50000',
       '1000000', '10000000', '5000', '100000000', '1000000000', '1000',
       '500000000', '50', '100', '500', '10', '1', '5', '0'], dtype=object)

In [125]:
# change data type of Installs:object to int
df_copy['Installs']=df_copy['Installs'].astype('int')

In [126]:
# change data type of Price:object to float
df_copy['Price']=df_copy['Price'].astype('float')

In [127]:
df_copy['Installs']

0           10000
1          500000
2         5000000
3        50000000
4          100000
           ...   
10836        5000
10837         100
10838        1000
10839        1000
10840    10000000
Name: Installs, Length: 10840, dtype: int32

In [128]:
df_copy['Price']

0        0.0
1        0.0
2        0.0
3        0.0
4        0.0
        ... 
10836    0.0
10837    0.0
10838    0.0
10839    0.0
10840    0.0
Name: Price, Length: 10840, dtype: float64

In [129]:
df_copy['Last Updated'].dtype

dtype('O')

In [130]:
df_copy['Last updated']=pd.to_datetime(df_copy['Last Updated'])
# convert into datetime.eg.January 7, 2018 to 1/7/2018

In [83]:
df_copy['Last updated']

0         7
1        15
2         1
3         8
4        20
         ..
10836    25
10837     6
10838    20
10839    19
10840    25
Name: Last updated, Length: 10840, dtype: int64

In [88]:
# create Day column
df_copy['Day']=df_copy['Last updated'].dt.day

In [89]:
# create month column
df_copy['month']=df_copy['Last updated'].dt.month

In [132]:
# create year column
df_copy['Year']=df_copy['Last updated'].dt.year

In [133]:
df_copy.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

   Reviews     Size  Installs  Type  Price Content Rating  \
0      159  19.0000     10000  Free    0.0       Everyone   
1      967  14.0000    500000  Free    0.0       Everyone   
2    87510   0.0087   5000000  Free    0.0       Everyone   
3   215644  25.0000  50000000  Free    0.0           Teen   
4      967   0.0028    100000  Free    0.0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver Last updated  Year  
0  4.0.3 and up   2018-01-07  2018  
1  4.0.3 and up   2018-01-15  2018  
2  4.0.3 and up   2018-08-01  2018  
3    4.2 and up   2018-06-08  2018  
4    4.4 and up   2018-06-20  2018

In [136]:
df_copy.to_csv('googleplaystore_cleaned.csv',index=False)
# save googleplaystore_cleaned file in csv format

In [137]:
pd.read_csv('googleplaystore_cleaned.csv')

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10835                                   Sya9a Maroc - FR               FAMILY   
10836                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10837                             Parkinson Exercices FR              MEDICAL   
10838                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10839      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating  Reviews     Size  Installs  Type  Price Content Rating  \
0         4.1      159  19.0000     10000  Free    0.0       Everyone   
1         3.9      967  14.0000    500000  Free    0.0       Everyone   
2         4.7    87510   0.0087   5000000  Free    0.0       Everyone   
3         4.5   215644  25.0000  50000000  Free    0.0           Teen   
4         4.3      967   0.0028    100000  Free    0.0       Everyone   
...       ...      ...      ...       ...   ...    ...            ...   
10835     4.5       38  53.0000      5000  Free    0.0       Everyone   
10836     5.0        4   0.0036       100  Free    0.0       Everyone   
10837     NaN        3   0.0095      1000  Free    0.0       Everyone   
10838     4.5      114      NaN      1000  Free    0.0     Mature 17+   
10839     4.5   398307  19.0000  10000000  Free    0.0       Everyone   

                          Genres      Last Updated         Current Ver  \
0                   Art & Design   January 7, 2018               1.0.0   
1      Art & Design;Pretend Play  January 15, 2018               2.0.0   
2                   Art & Design    August 1, 2018               1.2.4   
3                   Art & Design      June 8, 2018  Varies with device   
4        Art & Design;Creativity     June 20, 2018                 1.1   
...                          ...               ...                 ...   
10835                  Education     July 25, 2017                1.48   
10836                  Education      July 6, 2018                 1.0   
10837                    Medical  January 20, 2017                 1.0   
10838          Books & Reference  January 19, 2015  Varies with device   
10839                  Lifestyle     July 25, 2018  Varies with device   

              Android Ver Last updated  Year  
0            4.0.3 and up   2018-01-07  2018  
1            4.0.3 and up   2018-01-15  2018  
2            4.0.3 and up   2018-08-01  2018  
3              4.2 and up   2018-06-08  2018  
4              4.4 and up   2018-06-20  2018  
...                   ...          ...   ...  
10835          4.1 and up   2017-07-25  2017  
10836          4.1 and up   2018-07-06  2018  
10837          2.2 and up   2017-01-20  2017  
10838  Varies with device   2015-01-19  2015  
10839  Varies with device   2018-07-25  2018  

[10840 rows x 15 columns]

In [138]:
pd.read_csv('googleplaystore_cleaned.csv').info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10840 entries, 0 to 10839
Data columns (total 15 columns):
App               10840 non-null object
Category          10840 non-null object
Rating            9366 non-null float64
Reviews           10840 non-null int64
Size              9145 non-null float64
Installs          10840 non-null int64
Type              10839 non-null object
Price             10840 non-null float64
Content Rating    10840 non-null object
Genres            10840 non-null object
Last Updated      10840 non-null object
Current Ver       10832 non-null object
Android Ver       10838 non-null object
Last updated      10840 non-null object
Year              10840 non-null int64
dtypes: float64(3), int64(3), object(9)
memory usage: 1.2+ MB
